tensorboard/webapp/runs/store/runs_reducers.ts

```diff
@@ -295,6 +299,10 @@ const dataReducer: ActionReducer<RunsDataState, Action> = createReducer(
         }
       });
 
+      for (const run of groups.nonMatches) {
+        defaultRunColorForGroupBy.set(run.id, '');
```

Comment:
Allowing views to choose a 'default color' for non-matching runs is reasonable, however I would suggest making the state explicitly hold either a default color or `null` instead of empty strings.

Using an empty string means that all consumers of `getRunColorMap` **must** prefer doing `colorMap[runId] || defaultColor` instead of `colorMap[runId] ?? defaultColor`, if they want to show a default color for non-matching runs.  Making it a requirement for views to handle a `null` is slightly more obvious than an empty string I think, because TS will hopefully complain when a view tries to do `let x: string = colorMap[runId]` without handling the `null` case.

If we'd like to proceed with empty strings, could we
- Check that all consumers [1] handle the empty string, and update cases like ScalarCardContainer [2] to use `colorMap[runId] || defaultColor`
- Add comments in https://github.com/tensorflow/tensorboard/blob/master/tensorboard/webapp/runs/store/runs_types.ts#L50, explaining that the `defaultRunColorForGroupBy` field may contain empty string values for non-matching runs, and that `runColorOverrideForGroupBy` may not contain empty string values
- Update the comment on `getRunColorMap` [3], so that future consumers may realize that the map may contain empty string values

Or if you prefer using `null`, could we update the selector's comment to describe what a `null` value means?

Or if you prefer keeping a default color in the state, then it guarantees runs to have the same color across all views, and no further action needed.

Wdyt?

[1] https://cs.opensource.google/search?q=%22getRunColorMap)%22&sq=&ss=tensorflow%2Ftensorboard
[2] https://github.com/tensorflow/tensorboard/blob/master/tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts#L380
[3] https://github.com/tensorflow/tensorboard/blob/master/tensorboard/webapp/runs/store/runs_selectors.ts#L189

Reply:
I agree with empty string might cause unforeseen issues; null is not applicable cause `defaultRunColorForGroupBy` is `Map()` (type string is needed)
I modified it to delete the `run.id` from `defaultRunColorForGroupBy` thus all null check will work properly as before.

Reply:
Deleting sounds good, thanks!